### PR TITLE
Update Mage_Customer_Model_Convert_Parser_Customer method unparse()

### PR DESCRIPTION
**Pull request by matteorossi-thespacesm**
In class Mage_Customer_Model_Convert_Parser_Customer method unparse, a new CustomerAddress instance is fetched through a call to getCustomerAddressModel(), which in turn gets model instance from Mage::objects() cache.

Such pull request is proposing to bypass cache usage and create always a new CustomerAddress instance.

This is due to the fact that whenever a customer instance has empty field, such as custom customer fields which are not required, if the previous customer has a not empty value in such field, the previous value is exported even with the next customer, so it could be duplicated a number of times for any subsequent customer, until a new value is retrieved.

To test this, create a custom customer attribute with no required value, populate it for a customer and let it blank for subsequent customer. Create a dataflow including such attribute and try and export customers.

### CODE REVIEW

**Comment by tmotyl** [comment]
Hi @matteorossi-thespacesm 
Thanks for the PR!
Can you please describe the steps I can take to reproduce the issue this PR is fixing?
Thanks a lot!

**Comment by matteorossi-thespacesm** [comment]
Hello,

as I've stated in pull request description, you 'd have to:

  * populate a number of customers with not required attributes, some
    with random values, some leaving attribute blank
  * create a dataflow with attribute to export
  * export such dataflow

You'll see that previously populated values are propagated to customers 
which should have such attribute value blank.

Best Regards

Matteo Rossi



On 17/09/20 17:11, Tymoteusz Motylewski wrote:
>
> Hi @matteorossi-thespacesm <https://u11252189.ct.sendgrid.net/ls/click?upn=iGfHXqw3YnyGgJzxRg82lpQi3-2FIdYoL2ep8ELljvlhYp1goObVTuQQ02-2FTow4kD6jwCTQFIhvG0lWJcYTyms3g-3D-3DGmcs_U31jt5HgHBh7k3Grc5s4l7cGxXbxQyb1TgPRXLt0JL7Mc7-2FrK-2B4qA7lc1UzCwHCUMYHKlPa75wgzix4EN-2FpHLIziv4gnEOC3WQEPlR0rDUCjbNaOnQaFApUakgPABEYpQsCIeks5QYZHe9DKKViCRzb91FAUOcz8Q4FvRN9Qm0vKwq85af1pFuiSqnSTCC9mtxkJI3tKDUvIZFVMUSWmdzjv3hwPl8K5PZ-2FiZRae5-2Bq7NDxFELVCPds1iR5WbWRuewbJVeI-2B8lEx6-2F4EgfyJayq7hX9XgoJomT0zgndOf3A-3D>
> Thanks for the PR!
> Can you please describe the steps I can take to reproduce the issue 
> this PR is fixing?
> Thanks a lot!
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub 
> <https://u11252189.ct.sendgrid.net/ls/click?upn=iGfHXqw3YnyGgJzxRg82lqkOIAMKqS1ot41LAPB-2FNU2z6l126sOz1AsZpNYinh2WkBPBymIBv77RluD-2BUQaVqWQ-2BoyUK45QxGLReHKBJUzxBjF-2BEzQLz7zE66xYF5eVM8Pvs_U31jt5HgHBh7k3Grc5s4l7cGxXbxQyb1TgPRXLt0JL7Mc7-2FrK-2B4qA7lc1UzCwHCUMYHKlPa75wgzix4EN-2FpHLIziv4gnEOC3WQEPlR0rDUCjbNaOnQaFApUakgPABEYp-2BLHPE6-2BIAeR3w6R6H1NwjET3P1tqGYF7flkzaLxtP5cX-2FP2urUr020B8Ajy9-2BwM9kxvF-2BaCWf-2Bt8m8Pt08dPbEjkvKo4zIcrqivPJPKjKyGLEj-2B6FwyyM34kt3S0LAQ9hmH-2FamN44bsVmxRULIw2z-2Fnky1vTBenLj8pdYMLmpWQ-3D>, 
> or unsubscribe 
> <https://u11252189.ct.sendgrid.net/ls/click?upn=iGfHXqw3YnyGgJzxRg82lsr2W7ht2-2Bc9qUMXbZENzYwyAXNP-2BgvX-2F-2FTugQfCbDr78Uw9i3uvfEM8SB5DcmBrYQRW5CIZyHTzz77oPpkWPouTh6Gurj3hXURIqtyLjgj58kXJNvSbBp1rT3UzkrOPJw-3D-3DPu70_U31jt5HgHBh7k3Grc5s4l7cGxXbxQyb1TgPRXLt0JL7Mc7-2FrK-2B4qA7lc1UzCwHCUMYHKlPa75wgzix4EN-2FpHLIziv4gnEOC3WQEPlR0rDUCjbNaOnQaFApUakgPABEYpDXX2R-2F-2BflFTVUTVicgBg9YyMDFzmms8QrANdFO2lFXb0-2BrqEV13mZypGL6OML1VQg-2BdqotMxQ5lUkMggdcwecGEfO6QoRR3nXMe1ZtGk2MZRmwEVLDVCF4UTkPh1TL7rvYXnGYxct7hmeRRRMcoAalVG8gaLHhq73M7FeVX-2FE-2Fg-3D>.
>
-- 
Logo 	
Matteo Rossi
Full Stack Developer

Logo
Web 	+39 338 8306931‬

Email 	matteo.rossi@thespacesm.com <mailto:matteo.rossi@thespacesm.com>

Address 	Strada Cardio, 58 - Serravalle - 47899 San Marino

Web 	www.thespacesm.com <https://u11252189.ct.sendgrid.net/ls/click?upn=iGfHXqw3YnyGgJzxRg82lkUnY8vyNiVvqh3Bb0E2Wi6AfUj6LGxGErsRp9wu3ruQjfvk_U31jt5HgHBh7k3Grc5s4l7cGxXbxQyb1TgPRXLt0JL7Mc7-2FrK-2B4qA7lc1UzCwHCUMYHKlPa75wgzix4EN-2FpHLIziv4gnEOC3WQEPlR0rDUCjbNaOnQaFApUakgPABEYpXFRivBogYaD0hZI9deEzOB6bC5tcGrcgU9-2FHwBPE2xzagR6Ec6oFHXkXuvtYsy8hfhCDCJJbTdyVEagQm-2BCuV-2FZv83LJZ1EAP-2F-2BwM-2BWEmY6oTa3o5fidFR-2F9c1KqV0DjPRIPnTRVj4IgKtnmTpYJiH-2B72RbfbgjYQMsNltUYhXM-3D>

Logo

File: app/code/core/Mage/Customer/Model/Convert/Parser/Customer.php [hunk 1]
```diff
@@ -288,7 +288,8 @@ public function unparse()
             $defaultBillingId  = $customer->getDefaultBilling();
             $defaultShippingId = $customer->getDefaultShipping();
 
-            $customerAddress = $this->getCustomerAddressModel();
+            //$customerAddress = $this->getCustomerAddressModel();
```
Comment (fballiano) [inline app/code/core/Mage/Customer/Model/Convert/Parser/Customer.php:291]:
```suggestion
```

File: app/code/core/Mage/Customer/Model/Convert/Parser/Customer.php [hunk 2]
```diff
@@ -288,7 +288,7 @@ public function unparse()
             $defaultBillingId  = $customer->getDefaultBilling();
             $defaultShippingId = $customer->getDefaultShipping();
 
-            $customerAddress = $this->getCustomerAddressModel();
+            $customerAddress   = Mage::getModel('customer/address');
```
Comment (fballiano) [inline app/code/core/Mage/Customer/Model/Convert/Parser/Customer.php:291]:
```suggestion
            $customerAddress = Mage::getModel('customer/address');
```

File: app/code/core/Mage/Customer/Model/Convert/Parser/Customer.php [hunk 3]
```diff
@@ -84,6 +84,7 @@ public function getCustomerModel()
      * Retrieve customer address model cache
      *
      * @return Mage_Customer_Model_Address|object
+     * @deprecated
```
Comment (fballiano) [inline app/code/core/Mage/Customer/Model/Convert/Parser/Customer.php:87]: I've deprecated this method cause it's used only in this class (well, not anymore after this PR)

**Comment by fballiano** [comment]
I've tried some different things but I can't reproduce the problem, either on customer or customer_address, the data seems to be correct, any more info is apreciated but since it's a very old PR I'm closing it ATM.

**Pull request closed without merging**